snmp/tests/test_profiles.py

```diff
@@ -48,6 +48,20 @@
 pytestmark = pytest.mark.usefixtures("dd_environment")
 
 
+def _test_profile_syntax(check):
+    for name, profile in check.profiles.items():
+        try:
+            check._config.refresh_with_profile(profile)
+        except ConfigurationError as e:
+            pytest.fail("Profile `{}` is not configured correctly: {}".format(name, e))
```

Comment:
While I agree with Christine's suggestion to run this verification as part of individual profile checks, currently we verify all profiles each time which might not be ideal.

Perhaps as a follow up we can implement "enforce the `recording_name` and `.snmprec` to match the profile name" part, which would allow doing something like this…

```python
def run_profile_check(name):
    ...
    profile = check.profiles[name]
    verify_profile_syntax(name, profile)

    check.check(instance)
```

(Currently we can't because `recording_name != name` in general.)

Reply:
We decided to add this feature in a second PR